app/controllers/admin/admin_users_controller.rb

```diff
@@ -67,20 +65,47 @@ def update
     redirect_to request.referer || root_path
   end
 
+  def update_next_of_kin
+    @user = authorize User.find_by!(login: params[:user_login])
+    fnok = @user.fannish_next_of_kin
+    kin = User.find_by(login: params[:next_of_kin_name])
+    kin_email = params[:next_of_kin_email]
+
+    if kin.blank? && kin_email.blank?
+      if fnok.present?
+        fnok.destroy
+        flash[:notice] = ts("Fannish next of kin was removed.")
+      end
+      redirect_to admin_user_path(@user)
+      return
+    end
+
+    fnok = @user.build_fannish_next_of_kin(kin: kin, kin_email: kin_email)
+    if fnok.save
```

Comment:
I just did some experiments to check out how this works. Because `fannish_next_of_kin` is a `has_one`, calling `build_fannish_next_of_kin` will immediately delete the old fannish next-of-kin from the database, in a separate transaction from the call to `fnok.save`. So if the new fannish next-of-kin is invalid, it will result in the old next of kin being deleted without the new one being saved. Maybe there should be some kind of message telling the admin that the old next of kin has been removed? Or maybe this should be modified to avoid removing the old next of kin unless the new next of kin saves properly?

Reply:
Thanks for catching that! I'll change it to

> Or maybe this should be modified to avoid removing the old next of kin unless the new next of kin saves properly?